sos/report/plugins/__init__.py

```diff
@@ -1775,6 +1775,11 @@ def check_enabled(self):
         return True
 
     def _check_plugin_triggers(self, files, packages, commands, services):
+
+        if not any([files, packages, commands, services]):
```

Comment:
This would allow a file, package, command or service trigger to bypass an arch exclusion. Is that what we want, or should we go ahead and check other triggers iff the arch is a match?

I.e. should this be like:

```python
    if not self.check_is_architecture():
        return False

    # Other enablement checks
```

I.e. being on a not-matching architecture excludes us from running *regardless* of any other checks.

Reply:
If we have any files, package, commands, or services, then we skip past this and continue to the existing check which includes the arch check as an AND condition.

Though the simpler arch-only check is easier to understand at a glance...


Yeah, I'll change that over.

Reply:
Wait, actually - no that would still prevent plugins from enabling when they _only_ specify an arch. We'd pass the `check_is_architecture()` conditional, but then we'd fail the following (existing) conditional like we still do today, since all files, commands, packages, and services would evaluate `False`.

Reply:
Nm - `check_is_architecture()` is also (correctly) tested in the main `return` clause so the behaviour is that plugins with just an arch test rely only on that check, whereas those that also define package etc. triggers must match one or more of those AND the arch, which is exactly what we want.